src/web/components/StickyBottomBanner/ReaderRevenueBanner.tsx

```diff
@@ -157,96 +168,164 @@ export const canShow = async ({
 	return getBanner(
 		bannerPayload,
 		`${contributionsServiceUrl}/banner${queryString}`,
-	)
-		.then(checkForErrors)
-		.then((response) => response.json())
-		.then((json: { data?: any }) => {
+	).then((json: { data?: any }) => {
+		if (!json.data) {
+			if (
+				engagementBannerLastClosedAt &&
+				subscriptionBannerLastClosedAt
+			) {
+				setLocalNoBannerCachePeriod();
+			}
+			return { result: false };
+		}
+
+		const { module, meta } = json.data;
+
+		return { result: true, meta: { module, meta } };
+	});
+};
+
+export const canShowPuzzlesBanner: CanShowFunction = async ({
+	remoteBannerConfig,
+	isSignedIn,
+	asyncCountryCode,
+	contentType,
+	sectionName,
+	shouldHideReaderRevenue,
+	isMinuteArticle,
+	isPaidContent,
+	isSensitive,
+	tags,
+	contributionsServiceUrl,
+	alreadyVisitedCount,
+	engagementBannerLastClosedAt,
+	subscriptionBannerLastClosedAt,
+	section,
+}) => {
+	const isPuzzlesPage =
+		section === 'crosswords' ||
+		tags.some((tag) => tag.type === 'Series' && tag.title === 'Sudoku');
+
+	if (shouldHideReaderRevenue) {
+		// We never serve Reader Revenue banners in this case
+		return { result: false };
+	}
+
+	if (isPuzzlesPage && remoteBannerConfig) {
+		const countryCode = await asyncCountryCode;
+		const hasConsentedToArticleCounts = await getArticleCountConsent();
+		const bannerPayload = buildPayload({
+			isSignedIn,
+			countryCode,
+			contentType,
+			sectionName,
+			shouldHideReaderRevenue,
+			isMinuteArticle,
+			isPaidContent,
+			tags,
+			contributionsServiceUrl,
+			isSensitive,
+			alreadyVisitedCount,
+			engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt,
+			hasConsentedToArticleCounts,
+		});
+		return getBanner(
+			bannerPayload,
+			`${contributionsServiceUrl}/puzzles`,
+		).then((json: { data?: any }) => {
 			if (!json.data) {
-				if (
-					engagementBannerLastClosedAt &&
-					subscriptionBannerLastClosedAt
-				) {
-					setLocalNoBannerCachePeriod();
-				}
 				return { result: false };
 			}
 
 			const { module, meta } = json.data;
 
 			return { result: true, meta: { module, meta } };
 		});
+	}
+
+	return { result: false };
 };
 
-type Props = {
+export type BannerProps = {
 	meta: any;
 	module: { url: string; name: string; props: any[] };
 };
 
-export const ReaderRevenueBanner = ({ meta, module }: Props) => {
-	const [Banner, setBanner] = useState<React.FC>();
+const createReaderRevenueBannerComponent = (
+	componentTypeName: ReaderRevenueComponent,
+	displayEvent: string,
+): React.FC<BannerProps> => {
+	return ({ meta, module }: BannerProps) => {
```

Comment:
I would favour using a classic component pattern over a `HOC` pattern in this case, as technically, we are just passing through props rather than enhancing a component.

Example:
```
const BaseBannerComponent = ({
	componentTypeName: ReaderRevenueComponent,
	displayEvent: string,
	meta: any,
	module: { url: string; name: string; props: any[] },
}) => ...
```

It means that `ReaderRevenueBanner` could be used like:
```
export const ReaderRevenueBanner = ({ meta, module }) => 
	<BaseBannerComponent
		componentTypeName="ReaderRevenueComponent"
		displayEvent="subscription-banner : display"
		meta={meta}
		module={module}
	/>
```

Reply:
Makes sense! 👍